cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,325 @@
+                Fields.stringType(ARTIFACT_COL),
+                Fields.stringType(NAME_COL),
+                Fields.stringType(DESCRIPTION_COL),
+                Fields.longType(CREATED_COL),
+                Fields.longType(UPDATED_COL),
+                Fields.stringType(PIPELINE_COL),
+                Fields.intType(REVISION_COL))
+    .withPrimaryKeys(NAMESPACE_COL, GENERATION_COL, OWNER_COL, ID_COL)
+    .build();
+  private static final Gson GSON = new Gson();
+  private final TransactionRunner transactionRunner;
+
+  public DraftStore(TransactionRunner transactionRunner) {
+    this.transactionRunner = transactionRunner;
+  }
+
+  /**
+   * @param namespace the namespace to fetch the drafts from
+   * @param owner the id of the owner of the drafts
+   * @param sortRequest The sorting that should be applied to the results, pass null if no sorting is required.
+   * @return a list of drafts
+   */
+  public List<Draft> listDrafts(NamespaceSummary namespace, String owner,
+                                SortRequest sortRequest) throws RuntimeException {
+    AtomicReference<List<StructuredRow>> results = new AtomicReference<>();
+    List<Field<?>> prefix = new ArrayList<>(3);
+    prefix.add(Fields.stringField(NAMESPACE_COL, namespace.getName()));
+    prefix.add(Fields.longField(GENERATION_COL, namespace.getGeneration()));
+    prefix.add(Fields.stringField(OWNER_COL, owner));
+
+    try {
+      transactionRunner.run(context -> {
+        StructuredTable table = context.getTable(TABLE_ID);
+        Range range = Range.singleton(prefix);
+        List<StructuredRow> temp = new ArrayList<>();
+        try (CloseableIterator<StructuredRow> rowIter = table.scan(range, Integer.MAX_VALUE)) {
+          while (rowIter.hasNext()) {
+            temp.add(rowIter.next());
+          }
+        }
+        results.set(temp);
+      });
+    } catch (TransactionException e) {
+      handleError(e.getCause());
+    }
+
+    List<StructuredRow> sortedResults = doSort(results.get(), sortRequest);
+    return sortedResults.stream().map(this::fromRow).collect(Collectors.toList());
+  }
+
+  /**
+   * Helper method to apply the sorting onto a list of rows. Sorting needs to take place at the store-level so it can
+   * leverage the StructuredRow to enable sorting on any field.
+   *
+   * @param rows list of {@link StructuredRow} to be sorted
+   * @param sortRequest {@link SortRequest} describing the sort to be performed
+   * @return a sorted list of {@link StructuredRow}
+   */
+  private List<StructuredRow> doSort(List<StructuredRow> rows, SortRequest sortRequest) {
+    if (sortRequest == null) {
+      return rows;
+    }
+    String sortField = sortRequest.getFieldName();
+    Map<String, FieldType.Type> fieldTypes = TABLE_SPEC
+      .getFieldTypes()
+      .stream()
+      .collect(Collectors.toMap(FieldType::getName, FieldType::getType));
+    if (!fieldTypes.containsKey(sortField)) {
+      throw new IllegalArgumentException(
```

Comment:
this type of error should eventually result in a 400 response code. I think it's better to validate this earlier in the handler rather than here. Each handler method should be validating user input before passing it lower in the stack.

Reply:
Makes sense, moved the check into the handler. Also kept the check in the store incase someone attempts to call the `DraftService` or `DraftStore` directly